gcc/rust/parse/rust-parse-impl.h

```diff
@@ -6265,20 +6277,14 @@ Parser<ManagedTokenSource>::parse_path_generic_args ()
 	  && lexer.peek_token (1)->get_id () == EQUAL)
 	break;
 
-      auto type = parse_type (false);
-      if (type)
+      auto arg = parse_generic_arg ();
+      if (!arg.is_error ())
 	{
-	  type_args.emplace_back (std::move (type));
-	}
-      else
-	{
-	  auto const_generic_expr = parse_const_generic_expression ();
-	  if (const_generic_expr.is_error ())
-	    break;
-	  else
-	    const_args.emplace_back (std::move (const_generic_expr));
+	  generic_args.emplace_back (std::move (arg));
 	}
 
+      // FIXME: Do we need to break if we encounter an error?
```

Comment:
I think keep going because we might be able to recover the parser by reading until the delimiter for the arguments.